spec/controllers/shipments_controller_spec.rb

```diff
@@ -0,0 +1,134 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.describe ShipmentsController, type: :controller do
+  let(:successful_params) do
+    { origin: { "country" => "US", "city" => "Seattle", "state" => "WA", "zip" => 98105 },
+    destination: { country: "US", city: "Seattle", "state" => "WA", "zip" => 98105 },
+  packages: [{"weight" => 400, "dimensions" => "4, 5, 6" }, {"weight" => 100, "dimensions" => "6, 6, 8"} ]
+    }
+  end
+  let (:no_origin_params) do
+    { destination: { country: "US", city: "Seattle", "state" => "WA", "zip" => 98105 },
+     packages: [{"weight" => 400, "dimensions" => "4, 5, 6" }, {"weight" => 100, "dimensions" => "6, 6, 8"}] 
+   }
+  end
+  let(:no_origin_city_params) do
+    { origin: { "country" => "US", "state" => "WA", "zip" => 98105 },
+    destination: { country: "US", city: "Seattle", "state" => "WA", "zip" => 98105 },
+  packages: [{"weight" => 400, "dimensions" => "4, 5, 6" }, {"weight" => 100, "dimensions" => "6, 6, 8"} ]
+    }
+  end
+  let(:no_destination_params) do
+    { origin: { "country" => "US", "city" => "Seattle", "state" => "WA", "zip" => 98105 },
+     packages: [{"weight" => 400, "dimensions" => "4, 5, 6" }, {"weight" => 100, "dimensions" => "6, 6, 8"}]
+    }
+  end
+  let(:no_destination_city_params) do
+    { origin: { "country" => "US", "city" => "Seattle", "state" => "WA", "zip" => 98105 },
+    destination: { country: "US", "state" => "WA", "zip" => 98105 },
+  packages: [{"weight" => 400, "dimensions" => "4, 5, 6" }, {"weight" => 100, "dimensions" => "6, 6, 8"} ]
+    }
+  end
+  let(:no_destination_state_params) do
+    { origin: { "country" => "US", "city" => "Seattle", "state" => "WA", "zip" => 98105 },
+    destination: { country: "US", city: "Seattle", "zip" => 98105 },
+  packages: [{"weight" => 400, "dimensions" => "4, 5, 6" }, {"weight" => 100, "dimensions" => "6, 6, 8"} ]
+    }
+  end
+  let(:no_packages_params) do
+    { origin: { "country" => "US", "city" => "Seattle", "state" => "WA", "zip" => 98105 },
+    destination: { country: "US", city: "Seattle", "state" => "WA", "zip" => 98105 },
+    }
+  end
+  let(:empty_packages_params) do
+    { origin: { "country" => "US", "city" => "Seattle", "state" => "WA", "zip" => 98105 },
+    destination: { country: "US", city: "Seattle", state: "WA", zip: 98105 },
+    packages: []
+    }
+  end
+
+  describe "GET 'shipment'", :vcr do
+
+    it "is successful" do
```

Comment:
In what sense is this successful? It looks like an HTTP response code is being tested here, so it might be worth rewording this spec to say `it "returns a 200 HTTP status code"` or `it "returns a successful HTTP status"`. In a small project like Shipping Service, this might not matter. However, in a larger project, you may be testing for a whole [range of successful responses](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success).